hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -2057,4 +2057,40 @@ private V computeIfAbsentLocally(K key, Function<? super K, ? extends V> mapping
         }
     }
 
+    public V compute(@Nonnull K key, @Nonnull BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(key, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+
+        return computeLocally(key, remappingFunction);
+    }
+
+    private V computeLocally(K key,
+                             BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+
+        while (true) {
```

Comment:
I can't say I'm crazy about this approach because there is no guarantee for fairness and you could be stuck in this method for a long time.

Reply:
True, it can suffer from livelock but we need to update the value atomically by computing the new value locally. Do you think we should give up after some time?

Reply:
There's even a description of this problem in the javadoc and it mentions using serializable mapping functions instead - https://github.com/hazelcast/hazelcast/pull/17030/files#diff-881ec66ec2a01669cfd1f0952225df34R3101